bin/run_tests

```diff
@@ -7,56 +31,122 @@ top_dir=$(cd $(dirname $(dirname $0)) && pwd)
-PYTHONPATH=$PYTHONPATH:$top_dir/extensions/mktplace
-export PYTHONPATH
-cd extensions/mktplace
-nose2 -v unit
-export RUN_TEST_SUITES=1
-nose2 -v sawtooth_suites.ts_pr_mkt_poet0.Poet0MktTestSuite
-unset RUN_TEST_SUITES
-cd ../..
-
-PYTHONPATH=$top_dir
-PYTHONPATH=$PYTHONPATH:$top_dir/signing
-PYTHONPATH=$PYTHONPATH:$top_dir/signing/build/lib.linux-x86_64-2.7
-PYTHONPATH=$PYTHONPATH:$top_dir/core
-PYTHONPATH=$PYTHONPATH:$top_dir/validator
-PYTHONPATH=$PYTHONPATH:$top_dir/validator/build/lib.linux-x86_64-2.7
-PYTHONPATH=$PYTHONPATH:$top_dir/extensions/bond
-export PYTHONPATH
-cd extensions/bond
-nose2 -v
-cd ../..
+All() {
+	Core
+	Validator
+	Mktplace
+	Bond
+	Arcade
+}
+
+Core() {
+	PYTHONPATH=$top_dir/core
+	export PYTHONPATH
+	cd core
+	nose2 -v
+	cd ..
+}
+
+Validator() {
+	PYTHONPATH=$top_dir
+	PYTHONPATH=$PYTHONPATH:$top_dir/signing
+	PYTHONPATH=$PYTHONPATH:$top_dir/signing/build/lib.linux-x86_64-2.7
+	PYTHONPATH=$PYTHONPATH:$top_dir/core
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator/build/lib.linux-x86_64-2.7
+	export PYTHONPATH
+	cd validator
+	nose2 -v unit
+
+	# Disable nose2-3 within docker container until bugs can be resolved.
+	if [ "$DOCKER_CONTAINER_ID" = "" ]; then
+	    nose2-3 -v unit3
+	fi
+	
+	export RUN_TEST_SUITES=1
+	nose2 -v sawtooth_suites.ts_pr_dev_mode.DevModeTestSuite
+	nose2 -v sawtooth_suites.ts_pr_poet0.Poet0TestSuite
+	unset RUN_TEST_SUITES
+	cd ..
+}
+
+Mktplace() {
+	PYTHONPATH=$top_dir
+	PYTHONPATH=$PYTHONPATH:$top_dir/signing
+	PYTHONPATH=$PYTHONPATH:$top_dir/signing/build/lib.linux-x86_64-2.7
+	PYTHONPATH=$PYTHONPATH:$top_dir/core
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator/build/lib.linux-x86_64-2.7
+	PYTHONPATH=$PYTHONPATH:$top_dir/extensions/mktplace
+	export PYTHONPATH
+	cd extensions/mktplace
+	nose2 -v unit
+	export RUN_TEST_SUITES=1
+	nose2 -v sawtooth_suites.ts_pr_mkt_poet0.Poet0MktTestSuite
+	unset RUN_TEST_SUITES
+	cd ../..
+}
+
+Bond() {
+	PYTHONPATH=$top_dir
+	PYTHONPATH=$PYTHONPATH:$top_dir/signing
+	PYTHONPATH=$PYTHONPATH:$top_dir/signing/build/lib.linux-x86_64-2.7
+	PYTHONPATH=$PYTHONPATH:$top_dir/core
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator/build/lib.linux-x86_64-2.7
+	PYTHONPATH=$PYTHONPATH:$top_dir/extensions/bond
+	export PYTHONPATH
+	cd extensions/bond
+	nose2 -v
+	cd ../..
+}
+
+Arcade() { 
+	PYTHONPATH=$top_dir/core
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator
+	PYTHONPATH=$PYTHONPATH:$top_dir/validator/build/lib.linux-x86_64-2.7
+	PYTHONPATH=$PYTHONPATH:$top_dir/extensions/arcade
+	export PYTHONPATH
+	cd extensions/arcade
+	nose2 -v
+	cd ../..
+}
+
+MODULE="all"
+while getopts :m:h opt
+do
+    case $opt in
+      h)
+        usage
+        exit 0
+        ;;
+      m)
+        MODULE=$OPTARG
+        ;;
+      \?)
+        echo "Invalid option: -$OPTARG" >&2
+        exit 2
+        ;;
+    esac
+done
+
+case $MODULE in
+  all)
+    All
```

Comment:
Let's rename Core, Validator, etc. functions to 'lint_core', 'lint_validator', etc.  Let's also remove 'All' and just call each of these "lint_" functions directly from the case statement.